ramips:Add support for Phicomm K2P

981213 opened the pull request:
This commit adds device support for Phicomm K2P which uses MT7621A SoC.
It uses one MT7615D radio chip with DBDC mode enabled(This mode allow this single chip act as an 2x2 11n radio and an 2x2 11ac radio at the same time.)
However mt76 doesn't support it currently so there is no default packages defined.

mkresin commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -88,6 +88,13 @@ define Device/hc5962
 
+define Device/k2p
+  DTS := K2P
+  IMAGE_SIZE := 16056320
```
Your firmware partition is 16121856 bytes. Please use `15744k` here.

981213 replied:
A little question:How this size is calculated?

mkresin replied:
Not sure if I understand your question. The LEDE image is written to the firmware partition. Hence, the image can not be bigger than your firmware partition. Your firmware partition has a size of 0xf60000 => 16121856 bytes => 15744KByte.

981213 replied:
However I found that the variable ralink_default_fw_size_16M is 16121856(in target/linux/ramips/image/Makefile) which isn't equal to the partition size of firmware partition in the generic partition table of Ralink SDK(0x30000 for u-boot,0x10000 for u-boot-env 0x10000 for factory and 0xfb0000(or 16449536 bytes) for firmware).

mkresin replied:
As far as I know, the variable isn't related to the Ralink SDK. It is the default firmware partition size for a lot of the support boards with 16MByte flash. Even for your board ralink_default_fw_size_16M matches. Please use `IMAGE_SIZE := $(ralink_default_fw_size_16M)` instead of the former requested 15744k.

981213 commented:
I think It may be a little bit hard to write a short installation HOWTO.The factory firmware uses an RSA encrypted part to verify the uploaded binary file.Actually I used an SPI flash programmer to replace the firmware and U-boot.Maybe we could use an UART line to upload the firmware but I haven't try it.

There is another complicated way to replace the firmware which require users to generate a special config backup file from a third-party website and restore it to enable telnet.Then build an HTTP server on our computer to serve the binary file,use wget to download it to the router and use mtd command to write it.I'm wondering if describing this in the commit message is too long.And as it requires a third-party website and this way could be banned by the manufacture at any time, it's unreliable.

981213 commented:
I'll later describe the UART way in the commit message.
Shall I describe the complicated way I said above using a toh page on wiki or write it in the commit message?
@mkresin

mkresin commented:
> I'll later describe the UART way in the commit message.

Perfect. Can the stock u-boot used or is it necessary to replace the bootloader? Where to get the replacement boot loader from?

>  Shall I describe the complicated way I said above using a toh page on wiki or write it in the commit message?

At the wiki please. The used vulnerably might be closed with an update and/or the mentioned website might be offline.

981213 commented:
I think it's not necessary to replace the stock u-boot currently.

mkresin commented:
> I think it's not necessary to replace the stock u-boot currently.

Doesn't sound to me that you have tested it this way and the UART installation guide is more or less a good guess.

Would you please provide a bootlog - captured via serial console - which shows the installation via u-boot and the following LEDE boot.

981213 commented:
Here it is:
`
===================================================================
     		MT7621   stage1 code 10:33:55 (ASIC)
     		CPU=500000000 HZ BUS=166666666 HZ
==================================================================
Change MPLL source from XTAL to CR...
do MEMPLL setting..
MEMPLL Config : 0x11100000
3PLL mode + External loopback
=== XTAL-40Mhz === DDR-1200Mhz ===
PLL3 FB_DL: 0x4, 1/0 = 650/374 11000000
PLL2 FB_DL: 0x15, 1/0 = 617/407 55000000
PLL4 FB_DL: 0x1a, 1/0 = 542/482 69000000
do DDR setting..[01F40000]
Apply DDR3 Setting...(use customer AC)
          0    8   16   24   32   40   48   56   64   72   80   88   96  104  112  120
      --------------------------------------------------------------------------------
0000:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0001:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0002:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0003:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0004:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0005:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0006:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0007:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0008:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0009:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000A:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000B:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000C:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000D:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000E:|    0    0    0    0    0    0    0    0    0    0    1    1    1    1    1    1
000F:|    0    0    0    0    0    1    1    1    1    1    1    1    1    1    1    1
0010:|    1    1    1    1    1    1    1    1    1    1    0    0    0    0    0    0
0011:|    1    1    1    1    1    0    0    0    0    0    0    0    0    0    0    0
0012:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0013:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0014:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0015:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0016:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0017:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0018:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0019:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001A:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001B:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001C:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001D:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001E:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001F:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
rank 0 coarse = 15
rank 0 fine = 80
B:|    0    0    0    0    0    0    0    0    1    1    1    0    0    0    0    0
opt_dle value:9
DRAMC_R0DELDLY[018]=00002021
==================================================================
		RX	DQS perbit delay software calibration 
==================================================================
1.0-15 bit dq delay value
==================================================================
bit|     0  1  2  3  4  5  6  7  8  9
--------------------------------------
0 |    9 6 7 10 6 6 7 6 5 6 
10 |    7 8 7 9 6 8 
--------------------------------------

==================================================================
2.dqs window
x=pass dqs delay value (min~max)center 
y=0-7bit DQ of every group
input delay:DQS0 =33 DQS1 = 32
==================================================================
bit	DQS0	 bit      DQS1
0  (1~64)32  8  (1~59)30
1  (1~60)30  9  (1~61)31
2  (1~62)31  10  (1~60)30
3  (1~64)32  11  (1~61)31
4  (1~62)31  12  (1~64)32
5  (1~62)31  13  (2~62)32
6  (1~64)32  14  (1~63)32
7  (1~65)33  15  (1~64)32
==================================================================
3.dq delay value last
==================================================================
bit|    0  1  2  3  4  5  6  7  8   9
--------------------------------------
0 |    10 9 9 11 8 8 8 6 7 7 
10 |    9 9 7 9 6 8 
==================================================================
==================================================================
     TX  perbyte calibration 
==================================================================
DQS loop = 15, cmp_err_1 = ffff0000 
dqs_perbyte_dly.last_dqsdly_pass[0]=15,  finish count=1 
dqs_perbyte_dly.last_dqsdly_pass[1]=15,  finish count=2 
DQ loop=15, cmp_err_1 = ffff0000
dqs_perbyte_dly.last_dqdly_pass[0]=15,  finish count=1 
dqs_perbyte_dly.last_dqdly_pass[1]=15,  finish count=2 
byte:0, (DQS,DQ)=(8,8)
byte:1, (DQS,DQ)=(8,8)
20,data:88
[EMI] DRAMC calibration passed

===================================================================
     		MT7621   stage1 code done 
     		CPU=500000000 HZ BUS=166666666 HZ
===================================================================


U-Boot 1.1.3 ( - )

Board: Ralink APSoC DRAM:  128 MB
uboot gpio init : WPS / RESET pin
relocate_code Pointer at: 87fa8000

Config XHCI 40M PLL 
flash manufacture id: c2, device id 20 18
find flash: MX25L12805D
ra_spi_flash_size:0x1000000
*** Warning - bad CRC, using default environment

default load_addr 0x84000000
============================================ 
Ralink UBoot Version: 5.0.0.0
-------------------------------------------- 
ASIC MT7621A DualCore (MAC to MT7530 Mode)
DRAM_CONF_FROM: Auto-Detection 
DRAM_TYPE: DDR3 
DRAM bus: 16 bit
Xtal Mode=3 OCP Ratio=1/3
Flash component: SPI Flash
Date:  Time:
============================================ 
icache: sets:256, ways:4, linesz:32 ,total:32768
dcache: sets:256, ways:4, linesz:32 ,total:32768 

 ##### The CPU freq = 880 MHZ #### 
 estimate memory size =128 Mbytes
#Reset_MT7530
set LAN/WAN LLLLW

Press press hardware button for more than 4 seconds to run web failsafe mode



Catution: hardware button wasn't pressed or not long enough!
Continuing normal boot...


Please choose the operation: 
   1: Load system code to SDRAM via TFTP. 
   2: Load system code then write to Flash via TFTP. 
   3: Boot system code via Flash (default).
   4: Entr boot command line interface.
   6: Load all then write to Flash via TFTP. 
   7: Load Boot Loader code then write to Flash via Serial. 
   9: Load Boot Loader code then write to Flash via TFTP. 
default: 3

You choosed 2



   
2: System Load Linux Kernel then write to Flash via TFTP. 
 Warning!! Erase Linux in Flash then burn new one. Are you sure?(Y/N)
 Please Input new ones /or Ctrl-C to discard
	Input device IP (192.168.2.1) ==:192.168.2.1
	Input server IP (192.168.2.10) ==:192.168.2.22
	Input Linux Kernel filename () ==:fw.bin

 NetTxPacket = 0x87FE3480 

 KSEG1ADDR(NetTxPacket) = 0xA7FE3480 

 NetLoop,call eth_halt ! 

 NetLoop,call eth_init ! 
Trying Eth0 (10/100-M)

 Waitting for RX_DMA_BUSY status Start... done


 ETH_STATE_ACTIVE!! 
TFTP from server 192.168.2.22; our IP address is 192.168.2.1
Filename 'fw.bin'.

 TIMEOUT_COUNT=10,Load address: 0x84000000
Loading: checksum bad
checksum bad
Got ARP REPLY, set server/gtwy eth addr (00:07:e9:5e:33:bb)
Got it
#################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 ##################################################
done
Bytes transferred = 5243048 (5000a8 hex)
LoadAddr=84000000 NetBootFileXferSize= 005000a8
................................................................................
................................................................................
.
.
Done!
## Booting image at bc0a0000 ...
   Image Name:   MIPS LEDE Linux-4.9.37
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    1632839 Bytes =  1.6 MB
   Load Address: 80001000
   Entry Point:  80001000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK
No initrd
## Transferring control to Linux (at address 80001000) ...
## Giving linux memsize in MB, 128

Starting kernel ...

[    0.000000] Linux version 4.9.37 (user@GuoGuoPC.lan) (gcc version 5.4.0 (LEDE GCC 5.4.0 r4590-b4ce088f06) ) #0 SMP Tue Jul 18 12:34:38 2017
[    0.000000] SoC Type: MediaTek MT7621 ver:1 eco:3
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0001992f (MIPS 1004Kc)
[    0.000000] MIPS: machine is Phicomm K2P
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 08000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] VPE topology {2,2} total 4
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000]   HighMem  empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 12 pages/cpu @8110e000 s17584 r8192 d23376 u49152
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 32512
[    0.000000] Kernel command line: console=ttyS0,57600 rootfstype=squashfs,jffs2
[    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Writing ErrCtl register=00003015
[    0.000000] Readback ErrCtl register=00003015
[    0.000000] Memory: 124208K/131072K available (3871K kernel code, 201K rwdata, 888K rodata, 212K init, 267K bss, 6864K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000] NR_IRQS:256
[    0.000000] clocksource: GIC: mask: 0xffffffffffffffff max_cycles: 0xcaf478abb4, max_idle_ns: 440795247997 ns
[    0.000000] sched_clock: 32 bits at 100 Hz, resolution 10000000ns, wraps every 21474836475000000ns
[    0.010000] Calibrating delay loop... 586.13 BogoMIPS (lpj=2930688)
[    0.070000] pid_max: default: 32768 minimum: 301
[    0.080000] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.090000] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.100000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.100000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.100000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.100000] CPU1 revision is: 0001992f (MIPS 1004Kc)
[    0.190000] Synchronize counters for CPU 1: 
[    0.190000] done.
[    0.200000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.200000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.200000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.200000] CPU2 revision is: 0001992f (MIPS 1004Kc)
[    0.290000] Synchronize counters for CPU 2: 
[    0.290000] done.
[    0.300000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.300000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.300000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.300000] CPU3 revision is: 0001992f (MIPS 1004Kc)
[    0.390000] Synchronize counters for CPU 3: 
[    0.390000] done.
[    0.400000] Brought up 4 CPUs
[    0.410000] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.420000] futex hash table entries: 1024 (order: 3, 32768 bytes)
[    0.430000] pinctrl core: initialized pinctrl subsystem
[    0.440000] NET: Registered protocol family 16
[    0.480000] pull PCIe RST: RALINK_RSTCTRL = 4000000
[    0.780000] release PCIe RST: RALINK_RSTCTRL = 7000000
[    0.790000] ***** Xtal 40MHz *****
[    0.790000] release PCIe RST: RALINK_RSTCTRL = 7000000
[    0.800000] Port 0 N_FTS = 1b102800
[    0.800000] Port 1 N_FTS = 1b102800
[    0.810000] Port 2 N_FTS = 1b102800
[    1.960000] PCIE1 no card, disable it(RST&CLK)
[    1.970000] PCIE2 no card, disable it(RST&CLK)
[    1.970000]  -> 21007f2
[    1.980000] PCIE0 enabled
[    1.980000] PCI host bridge /pcie@1e140000 ranges:
[    1.990000]  MEM 0x0000000060000000..0x000000006fffffff
[    2.000000]   IO 0x000000001e160000..0x000000001e16ffff
[    2.010000] PCI coherence region base: 0x60000000, mask/settings: 0xf0000002
[    2.050000] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    2.060000] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    2.070000] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    2.080000] PCI host bridge to bus 0000:00
[    2.080000] pci_bus 0000:00: root bus resource [mem 0x60000000-0x6fffffff]
[    2.090000] pci_bus 0000:00: root bus resource [io  0xffffffff]
[    2.100000] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    2.110000] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    2.120000] pci 0000:00:00.0: bridge configuration invalid ([bus 00-00]), reconfiguring
[    2.130000] pci 0000:00:00.0: BAR 0: no space for [mem size 0x80000000]
[    2.140000] pci 0000:00:00.0: BAR 0: failed to assign [mem size 0x80000000]
[    2.150000] pci 0000:00:00.0: BAR 8: assigned [mem 0x60000000-0x600fffff]
[    2.160000] pci 0000:00:00.0: BAR 1: assigned [mem 0x60100000-0x6010ffff]
[    2.170000] pci 0000:01:00.0: BAR 0: assigned [mem 0x60000000-0x600fffff 64bit]
[    2.180000] pci 0000:00:00.0: PCI bridge to [bus 01]
[    2.180000] pci 0000:00:00.0:   bridge window [mem 0x60000000-0x600fffff]
[    2.190000] BAR0 at slot 0 = 0
[    2.190000] bus=0x0, slot = 0x0
[    2.200000] bus=0x1, slot = 0x0, irq=0x0
[    2.200000] clocksource: Switched to clocksource GIC
[    2.220000] NET: Registered protocol family 2
[    2.230000] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    2.240000] TCP bind hash table entries: 1024 (order: 1, 8192 bytes)
[    2.250000] TCP: Hash tables configured (established 1024 bind 1024)
[    2.260000] UDP hash table entries: 256 (order: 1, 8192 bytes)
[    2.280000] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes)
[    2.290000] NET: Registered protocol family 1
[    2.300000] Crashlog allocated RAM at address 0x3f00000
[    2.310000] workingset: timestamp_bits=30 max_order=15 bucket_order=0
[    2.330000] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    2.340000] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    2.360000] random: fast init done
[    2.360000] io scheduler noop registered
[    2.360000] io scheduler deadline registered (default)
[    2.360000] Serial: 8250/16550 driver, 3 ports, IRQ sharing disabled
[    2.370000] console [ttyS0] disabled
[    2.410000] 1e000c00.uartlite: ttyS0 at MMIO 0x1e000c00 (irq = 17, base_baud = 3125000) is a 16550A
[    2.420000] console [ttyS0] enabled
[    2.420000] console [ttyS0] enabled
[    2.440000] bootconsole [early0] disabled
[    2.440000] bootconsole [early0] disabled
[    2.460000] MediaTek Nand driver init, version v2.1 Fix AHB virt2phys error
[    2.470000] spi-mt7621 1e000b00.spi: sys_freq: 50000000
[    2.480000] m25p80 spi32766.0: using chunked io (size=32)
[    2.490000] m25p80 spi32766.0: mx25l12805d (16384 Kbytes)
[    2.500000] 5 ofpart partitions found on MTD device spi32766.0
[    2.510000] Creating 5 MTD partitions on "spi32766.0":
[    2.520000] 0x000000000000-0x000000030000 : "u-boot"
[    2.540000] 0x000000030000-0x000000040000 : "u-boot-env"
[    2.550000] 0x000000040000-0x000000050000 : "factory"
[    2.560000] 0x000000050000-0x0000000a0000 : "permanent_config"
[    2.570000] 0x0000000a0000-0x000001000000 : "firmware"
[    2.620000] 2 uimage-fw partitions found on MTD device firmware
[    2.630000] 0x0000000a0000-0x00000022ea87 : "kernel"
[    2.640000] 0x00000022ea87-0x000001000000 : "rootfs"
[    2.650000] mtd: device 6 (rootfs) set to be root filesystem
[    2.660000] 1 squashfs-split partitions found on MTD device rootfs
[    2.680000] 0x000000570000-0x000001000000 : "rootfs_data"
[    2.690000] libphy: Fixed MDIO Bus: probed
[    2.700000] netif_napi_add() called with weight 128 on device eth%d
[    2.720000] libphy: mdio: probed
[    4.130000] mtk_soc_eth 1e100000.ethernet: loaded mt7530 driver
[    4.140000] mtk_soc_eth 1e100000.ethernet eth0: mediatek frame engine at 0xbe100000, irq 19
[    4.160000] NET: Registered protocol family 10
[    4.170000] NET: Registered protocol family 17
[    4.180000] 8021q: 802.1Q VLAN Support v1.8
[    4.190000] hctosys: unable to open rtc device (rtc0)
[    4.200000] VFS: Mounted root (squashfs filesystem) readonly on device 31:6.
[    4.220000] Freeing unused kernel memory: 212K (804db000 - 80510000)
[    4.230000] This architecture does not have kernel memory protection.
[    4.590000] random: crng init done
[    4.750000] init: Console is alive
[    5.230000] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    5.340000] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    5.360000] init: - preinit -
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    6.040000] mtk_soc_eth 1e100000.ethernet eth0: port 2 link up
[    8.050000] mount_root: jffs2 not ready yet, using temporary tmpfs overlay
[    8.070000] urandom-seed: Seed file not found (/etc/urandom.seed)
[    8.210000] procd: - early -
[    8.890000] procd: - ubus -
[    8.950000] procd: - init -
Please press Enter to activate this console.
[    9.120000] kmodloader: loading kernel modules from /etc/modules.d/*
[    9.140000] ralink-dep:module loaded.
[    9.150000] ip6_tables: (C) 2000-2006 Netfilter Core Team
[   10.690000] mt7615_dbdc: module license 'Proprietary' taints kernel.
[   10.700000] Disabling lock debugging due to kernel taint
[   10.780000] register mt_drv
[   10.800000] 
[   10.800000] 
[   10.800000] === pAd = c0701000, size = 3631384 ===
[   10.800000] 
[   10.820000] <-- RTMPAllocAdapterBlock, Status=0
[   10.830000] get_wdev_by_idx: invalid idx(0)
[   10.840000] pAd->PciHif.CSRBaseAddress =0xc0600000, csr_addr=0xc0600000!
[   10.850000] get_wdev_by_idx: invalid idx(0)
[   10.860000] RTMPInitPCIeDevice():device_id=0x7615
[   10.870000] DriverOwn()::Try to Clear FW Own...
[   11.180000] DriverOwn()::Success to clear FW Own
[   11.190000] mt_pci_chip_cfg(): HWVer=0x8a10, FWVer=0x8a10, pAd->ChipID=0x7615
[   11.200000] mt_pci_chip_cfg(): HIF_SYS_REV=0x76150001
[   11.210000] RtmpChipOpsHook(493): Not support for HIF_MT yet! MACVersion=0x0
[   11.230000] mt7615_init()-->
[   11.230000] Use 1st ePAeLNA default bin.
[   11.240000] Use 0st /etc_ro/wlan/MT7615E_EEPROM1.bin default bin.
[   11.250000] <--mt7615_init()
[   11.260000] ChipOpsMCUHook
[   11.270000] cut_through_token_list_init(): TokenList inited done!id_head/tail=0/4096
[   11.280000] cut_through_token_list_init(): 872e6488,872e6488
[   11.300000] cut_through_token_list_init(): TokenList inited done!id_head/tail=0/4096
[   11.310000] cut_through_token_list_init(): 872e6498,872e6498
[   11.320000] <-- RTMPAllocTxRxRingMemory, Status=0
[   11.330000] get_wdev_by_idx: invalid idx(0)
[   11.340000] get_wdev_by_idx: invalid idx(0)
[   11.350000] get_wdev_by_idx: invalid idx(0)
[   11.360000] get_wdev_by_idx: invalid idx(0)
[   11.370000] get_wdev_by_idx: invalid idx(0)
[   11.370000] get_wdev_by_idx: invalid idx(0)
[   11.380000] get_wdev_by_idx: invalid idx(0)
[   11.390000] pci probe count=1
[   11.400000] ip_tables: (C) 2000-2006 Netfilter Core Team
[   11.420000] nf_conntrack version 0.5.0 (2048 buckets, 8192 max)
[   11.460000] xt_time: kernel timezone is -0000
[   11.480000] PPP generic driver version 2.4.2
[   11.490000] NET: Registered protocol family 24
[   11.500000] kmodloader: done loading kernel modules from /etc/modules.d/*
[   14.730000] get_wdev_by_idx: invalid idx(0)
[   17.290000] get_wdev_by_idx: invalid idx(0)
[   17.300000] get_wdev_by_idx: invalid idx(0)
[   17.490000] jffs2_scan_eraseblock(): End of filesystem marker found at 0x0
[   17.500000] jffs2_build_filesystem(): unlocking the mtd device... [   17.520000] done.
[   17.520000] jffs2_build_filesystem(): erasing all blocks after the end marker... [   17.660000] br-lan: port 1(eth0.1) entered blocking state
[   17.670000] br-lan: port 1(eth0.1) entered disabled state
[   17.680000] device eth0.1 entered promiscuous mode
[   17.690000] device eth0 entered promiscuous mode
[   17.700000] br-lan: port 1(eth0.1) entered blocking state
[   17.710000] br-lan: port 1(eth0.1) entered forwarding state
[   18.160000] get_wdev_by_idx: invalid idx(0)
[   18.170000] get_wdev_by_idx: invalid idx(0)
[   21.310000] get_wdev_by_idx: invalid idx(0)
[   21.620000] get_wdev_by_idx: invalid idx(0)
[   67.360000] done.
[   67.360000] jffs2: notice: (855) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
`

981213 commented:
Ah...I may have some troble using the markdown on GitHub,sorry...

981213 commented:
It should be OK this time...

```

===================================================================
     		MT7621   stage1 code 10:33:55 (ASIC)
     		CPU=500000000 HZ BUS=166666666 HZ
==================================================================
Change MPLL source from XTAL to CR...
do MEMPLL setting..
MEMPLL Config : 0x11100000
3PLL mode + External loopback
=== XTAL-40Mhz === DDR-1200Mhz ===
PLL3 FB_DL: 0x4, 1/0 = 650/374 11000000
PLL2 FB_DL: 0x15, 1/0 = 617/407 55000000
PLL4 FB_DL: 0x1a, 1/0 = 542/482 69000000
do DDR setting..[01F40000]
Apply DDR3 Setting...(use customer AC)
          0    8   16   24   32   40   48   56   64   72   80   88   96  104  112  120
      --------------------------------------------------------------------------------
0000:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0001:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0002:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0003:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0004:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0005:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0006:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0007:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0008:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0009:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000A:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000B:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000C:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000D:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
000E:|    0    0    0    0    0    0    0    0    0    0    1    1    1    1    1    1
000F:|    0    0    0    0    0    1    1    1    1    1    1    1    1    1    1    1
0010:|    1    1    1    1    1    1    1    1    1    1    0    0    0    0    0    0
0011:|    1    1    1    1    1    0    0    0    0    0    0    0    0    0    0    0
0012:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0013:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0014:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0015:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0016:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0017:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0018:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
0019:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001A:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001B:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001C:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001D:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001E:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
001F:|    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
rank 0 coarse = 15
rank 0 fine = 80
B:|    0    0    0    0    0    0    0    0    1    1    1    0    0    0    0    0
opt_dle value:9
DRAMC_R0DELDLY[018]=00002021
==================================================================
		RX	DQS perbit delay software calibration 
==================================================================
1.0-15 bit dq delay value
==================================================================
bit|     0  1  2  3  4  5  6  7  8  9
--------------------------------------
0 |    9 6 7 10 6 6 7 6 5 6 
10 |    7 8 7 9 6 8 
--------------------------------------

==================================================================
2.dqs window
x=pass dqs delay value (min~max)center 
y=0-7bit DQ of every group
input delay:DQS0 =33 DQS1 = 32
==================================================================
bit	DQS0	 bit      DQS1
0  (1~64)32  8  (1~59)30
1  (1~60)30  9  (1~61)31
2  (1~62)31  10  (1~60)30
3  (1~64)32  11  (1~61)31
4  (1~62)31  12  (1~64)32
5  (1~62)31  13  (2~62)32
6  (1~64)32  14  (1~63)32
7  (1~65)33  15  (1~64)32
==================================================================
3.dq delay value last
==================================================================
bit|    0  1  2  3  4  5  6  7  8   9
--------------------------------------
0 |    10 9 9 11 8 8 8 6 7 7 
10 |    9 9 7 9 6 8 
==================================================================
==================================================================
     TX  perbyte calibration 
==================================================================
DQS loop = 15, cmp_err_1 = ffff0000 
dqs_perbyte_dly.last_dqsdly_pass[0]=15,  finish count=1 
dqs_perbyte_dly.last_dqsdly_pass[1]=15,  finish count=2 
DQ loop=15, cmp_err_1 = ffff0000
dqs_perbyte_dly.last_dqdly_pass[0]=15,  finish count=1 
dqs_perbyte_dly.last_dqdly_pass[1]=15,  finish count=2 
byte:0, (DQS,DQ)=(8,8)
byte:1, (DQS,DQ)=(8,8)
20,data:88
[EMI] DRAMC calibration passed

===================================================================
     		MT7621   stage1 code done 
     		CPU=500000000 HZ BUS=166666666 HZ
===================================================================


U-Boot 1.1.3 ( - )

Board: Ralink APSoC DRAM:  128 MB
uboot gpio init : WPS / RESET pin
relocate_code Pointer at: 87fa8000

Config XHCI 40M PLL 
flash manufacture id: c2, device id 20 18
find flash: MX25L12805D
ra_spi_flash_size:0x1000000
*** Warning - bad CRC, using default environment

default load_addr 0x84000000
============================================ 
Ralink UBoot Version: 5.0.0.0
-------------------------------------------- 
ASIC MT7621A DualCore (MAC to MT7530 Mode)
DRAM_CONF_FROM: Auto-Detection 
DRAM_TYPE: DDR3 
DRAM bus: 16 bit
Xtal Mode=3 OCP Ratio=1/3
Flash component: SPI Flash
Date:  Time:
============================================ 
icache: sets:256, ways:4, linesz:32 ,total:32768
dcache: sets:256, ways:4, linesz:32 ,total:32768 

 ##### The CPU freq = 880 MHZ #### 
 estimate memory size =128 Mbytes
#Reset_MT7530
set LAN/WAN LLLLW

Press press hardware button for more than 4 seconds to run web failsafe mode



Catution: hardware button wasn't pressed or not long enough!
Continuing normal boot...


Please choose the operation: 
   1: Load system code to SDRAM via TFTP. 
   2: Load system code then write to Flash via TFTP. 
   3: Boot system code via Flash (default).
   4: Entr boot command line interface.
   6: Load all then write to Flash via TFTP. 
   7: Load Boot Loader code then write to Flash via Serial. 
   9: Load Boot Loader code then write to Flash via TFTP. 
default: 3

You choosed 2



   
2: System Load Linux Kernel then write to Flash via TFTP. 
 Warning!! Erase Linux in Flash then burn new one. Are you sure?(Y/N)
 Please Input new ones /or Ctrl-C to discard
	Input device IP (192.168.2.1) ==:192.168.2.1
	Input server IP (192.168.2.10) ==:192.168.2.22
	Input Linux Kernel filename () ==:fw.bin

 NetTxPacket = 0x87FE3480 

 KSEG1ADDR(NetTxPacket) = 0xA7FE3480 

 NetLoop,call eth_halt ! 

 NetLoop,call eth_init ! 
Trying Eth0 (10/100-M)

 Waitting for RX_DMA_BUSY status Start... done


 ETH_STATE_ACTIVE!! 
TFTP from server 192.168.2.22; our IP address is 192.168.2.1
Filename 'fw.bin'.

 TIMEOUT_COUNT=10,Load address: 0x84000000
Loading: checksum bad
checksum bad
Got ARP REPLY, set server/gtwy eth addr (00:07:e9:5e:33:bb)
Got it
#################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 #################################################################
	 ##################################################
done
Bytes transferred = 5243048 (5000a8 hex)
LoadAddr=84000000 NetBootFileXferSize= 005000a8
................................................................................
................................................................................
.
.
Done!
## Booting image at bc0a0000 ...
   Image Name:   MIPS LEDE Linux-4.9.37
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    1632839 Bytes =  1.6 MB
   Load Address: 80001000
   Entry Point:  80001000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK
No initrd
## Transferring control to Linux (at address 80001000) ...
## Giving linux memsize in MB, 128

Starting kernel ...

[    0.000000] Linux version 4.9.37 (user@GuoGuoPC.lan) (gcc version 5.4.0 (LEDE GCC 5.4.0 r4590-b4ce088f06) ) #0 SMP Tue Jul 18 12:34:38 2017
[    0.000000] SoC Type: MediaTek MT7621 ver:1 eco:3
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0001992f (MIPS 1004Kc)
[    0.000000] MIPS: machine is Phicomm K2P
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 08000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] VPE topology {2,2} total 4
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000]   HighMem  empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 12 pages/cpu @8110e000 s17584 r8192 d23376 u49152
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 32512
[    0.000000] Kernel command line: console=ttyS0,57600 rootfstype=squashfs,jffs2
[    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Writing ErrCtl register=00003015
[    0.000000] Readback ErrCtl register=00003015
[    0.000000] Memory: 124208K/131072K available (3871K kernel code, 201K rwdata, 888K rodata, 212K init, 267K bss, 6864K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000] NR_IRQS:256
[    0.000000] clocksource: GIC: mask: 0xffffffffffffffff max_cycles: 0xcaf478abb4, max_idle_ns: 440795247997 ns
[    0.000000] sched_clock: 32 bits at 100 Hz, resolution 10000000ns, wraps every 21474836475000000ns
[    0.010000] Calibrating delay loop... 586.13 BogoMIPS (lpj=2930688)
[    0.070000] pid_max: default: 32768 minimum: 301
[    0.080000] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.090000] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.100000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.100000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.100000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.100000] CPU1 revision is: 0001992f (MIPS 1004Kc)
[    0.190000] Synchronize counters for CPU 1: 
[    0.190000] done.
[    0.200000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.200000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.200000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.200000] CPU2 revision is: 0001992f (MIPS 1004Kc)
[    0.290000] Synchronize counters for CPU 2: 
[    0.290000] done.
[    0.300000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.300000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.300000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.300000] CPU3 revision is: 0001992f (MIPS 1004Kc)
[    0.390000] Synchronize counters for CPU 3: 
[    0.390000] done.
[    0.400000] Brought up 4 CPUs
[    0.410000] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.420000] futex hash table entries: 1024 (order: 3, 32768 bytes)
[    0.430000] pinctrl core: initialized pinctrl subsystem
[    0.440000] NET: Registered protocol family 16
[    0.480000] pull PCIe RST: RALINK_RSTCTRL = 4000000
[    0.780000] release PCIe RST: RALINK_RSTCTRL = 7000000
[    0.790000] ***** Xtal 40MHz *****
[    0.790000] release PCIe RST: RALINK_RSTCTRL = 7000000
[    0.800000] Port 0 N_FTS = 1b102800
[    0.800000] Port 1 N_FTS = 1b102800
[    0.810000] Port 2 N_FTS = 1b102800
[    1.960000] PCIE1 no card, disable it(RST&CLK)
[    1.970000] PCIE2 no card, disable it(RST&CLK)
[    1.970000]  -> 21007f2
[    1.980000] PCIE0 enabled
[    1.980000] PCI host bridge /pcie@1e140000 ranges:
[    1.990000]  MEM 0x0000000060000000..0x000000006fffffff
[    2.000000]   IO 0x000000001e160000..0x000000001e16ffff
[    2.010000] PCI coherence region base: 0x60000000, mask/settings: 0xf0000002
[    2.050000] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    2.060000] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    2.070000] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    2.080000] PCI host bridge to bus 0000:00
[    2.080000] pci_bus 0000:00: root bus resource [mem 0x60000000-0x6fffffff]
[    2.090000] pci_bus 0000:00: root bus resource [io  0xffffffff]
[    2.100000] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    2.110000] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    2.120000] pci 0000:00:00.0: bridge configuration invalid ([bus 00-00]), reconfiguring
[    2.130000] pci 0000:00:00.0: BAR 0: no space for [mem size 0x80000000]
[    2.140000] pci 0000:00:00.0: BAR 0: failed to assign [mem size 0x80000000]
[    2.150000] pci 0000:00:00.0: BAR 8: assigned [mem 0x60000000-0x600fffff]
[    2.160000] pci 0000:00:00.0: BAR 1: assigned [mem 0x60100000-0x6010ffff]
[    2.170000] pci 0000:01:00.0: BAR 0: assigned [mem 0x60000000-0x600fffff 64bit]
[    2.180000] pci 0000:00:00.0: PCI bridge to [bus 01]
[    2.180000] pci 0000:00:00.0:   bridge window [mem 0x60000000-0x600fffff]
[    2.190000] BAR0 at slot 0 = 0
[    2.190000] bus=0x0, slot = 0x0
[    2.200000] bus=0x1, slot = 0x0, irq=0x0
[    2.200000] clocksource: Switched to clocksource GIC
[    2.220000] NET: Registered protocol family 2
[    2.230000] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    2.240000] TCP bind hash table entries: 1024 (order: 1, 8192 bytes)
[    2.250000] TCP: Hash tables configured (established 1024 bind 1024)
[    2.260000] UDP hash table entries: 256 (order: 1, 8192 bytes)
[    2.280000] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes)
[    2.290000] NET: Registered protocol family 1
[    2.300000] Crashlog allocated RAM at address 0x3f00000
[    2.310000] workingset: timestamp_bits=30 max_order=15 bucket_order=0
[    2.330000] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    2.340000] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    2.360000] random: fast init done
[    2.360000] io scheduler noop registered
[    2.360000] io scheduler deadline registered (default)
[    2.360000] Serial: 8250/16550 driver, 3 ports, IRQ sharing disabled
[    2.370000] console [ttyS0] disabled
[    2.410000] 1e000c00.uartlite: ttyS0 at MMIO 0x1e000c00 (irq = 17, base_baud = 3125000) is a 16550A
[    2.420000] console [ttyS0] enabled
[    2.420000] console [ttyS0] enabled
[    2.440000] bootconsole [early0] disabled
[    2.440000] bootconsole [early0] disabled
[    2.460000] MediaTek Nand driver init, version v2.1 Fix AHB virt2phys error
[    2.470000] spi-mt7621 1e000b00.spi: sys_freq: 50000000
[    2.480000] m25p80 spi32766.0: using chunked io (size=32)
[    2.490000] m25p80 spi32766.0: mx25l12805d (16384 Kbytes)
[    2.500000] 5 ofpart partitions found on MTD device spi32766.0
[    2.510000] Creating 5 MTD partitions on "spi32766.0":
[    2.520000] 0x000000000000-0x000000030000 : "u-boot"
[    2.540000] 0x000000030000-0x000000040000 : "u-boot-env"
[    2.550000] 0x000000040000-0x000000050000 : "factory"
[    2.560000] 0x000000050000-0x0000000a0000 : "permanent_config"
[    2.570000] 0x0000000a0000-0x000001000000 : "firmware"
[    2.620000] 2 uimage-fw partitions found on MTD device firmware
[    2.630000] 0x0000000a0000-0x00000022ea87 : "kernel"
[    2.640000] 0x00000022ea87-0x000001000000 : "rootfs"
[    2.650000] mtd: device 6 (rootfs) set to be root filesystem
[    2.660000] 1 squashfs-split partitions found on MTD device rootfs
[    2.680000] 0x000000570000-0x000001000000 : "rootfs_data"
[    2.690000] libphy: Fixed MDIO Bus: probed
[    2.700000] netif_napi_add() called with weight 128 on device eth%d
[    2.720000] libphy: mdio: probed
[    4.130000] mtk_soc_eth 1e100000.ethernet: loaded mt7530 driver
[    4.140000] mtk_soc_eth 1e100000.ethernet eth0: mediatek frame engine at 0xbe100000, irq 19
[    4.160000] NET: Registered protocol family 10
[    4.170000] NET: Registered protocol family 17
[    4.180000] 8021q: 802.1Q VLAN Support v1.8
[    4.190000] hctosys: unable to open rtc device (rtc0)
[    4.200000] VFS: Mounted root (squashfs filesystem) readonly on device 31:6.
[    4.220000] Freeing unused kernel memory: 212K (804db000 - 80510000)
[    4.230000] This architecture does not have kernel memory protection.
[    4.590000] random: crng init done
[    4.750000] init: Console is alive
[    5.230000] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    5.340000] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    5.360000] init: - preinit -
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    6.040000] mtk_soc_eth 1e100000.ethernet eth0: port 2 link up
[    8.050000] mount_root: jffs2 not ready yet, using temporary tmpfs overlay
[    8.070000] urandom-seed: Seed file not found (/etc/urandom.seed)
[    8.210000] procd: - early -
[    8.890000] procd: - ubus -
[    8.950000] procd: - init -
Please press Enter to activate this console.
[    9.120000] kmodloader: loading kernel modules from /etc/modules.d/*
[    9.140000] ralink-dep:module loaded.
[    9.150000] ip6_tables: (C) 2000-2006 Netfilter Core Team
[   10.690000] mt7615_dbdc: module license 'Proprietary' taints kernel.
[   10.700000] Disabling lock debugging due to kernel taint
[   10.780000] register mt_drv
[   10.800000] 
[   10.800000] 
[   10.800000] === pAd = c0701000, size = 3631384 ===
[   10.800000] 
[   10.820000] <-- RTMPAllocAdapterBlock, Status=0
[   10.830000] get_wdev_by_idx: invalid idx(0)
[   10.840000] pAd->PciHif.CSRBaseAddress =0xc0600000, csr_addr=0xc0600000!
[   10.850000] get_wdev_by_idx: invalid idx(0)
[   10.860000] RTMPInitPCIeDevice():device_id=0x7615
[   10.870000] DriverOwn()::Try to Clear FW Own...
[   11.180000] DriverOwn()::Success to clear FW Own
[   11.190000] mt_pci_chip_cfg(): HWVer=0x8a10, FWVer=0x8a10, pAd->ChipID=0x7615
[   11.200000] mt_pci_chip_cfg(): HIF_SYS_REV=0x76150001
[   11.210000] RtmpChipOpsHook(493): Not support for HIF_MT yet! MACVersion=0x0
[   11.230000] mt7615_init()-->
[   11.230000] Use 1st ePAeLNA default bin.
[   11.240000] Use 0st /etc_ro/wlan/MT7615E_EEPROM1.bin default bin.
[   11.250000] <--mt7615_init()
[   11.260000] ChipOpsMCUHook
[   11.270000] cut_through_token_list_init(): TokenList inited done!id_head/tail=0/4096
[   11.280000] cut_through_token_list_init(): 872e6488,872e6488
[   11.300000] cut_through_token_list_init(): TokenList inited done!id_head/tail=0/4096
[   11.310000] cut_through_token_list_init(): 872e6498,872e6498
[   11.320000] <-- RTMPAllocTxRxRingMemory, Status=0
[   11.330000] get_wdev_by_idx: invalid idx(0)
[   11.340000] get_wdev_by_idx: invalid idx(0)
[   11.350000] get_wdev_by_idx: invalid idx(0)
[   11.360000] get_wdev_by_idx: invalid idx(0)
[   11.370000] get_wdev_by_idx: invalid idx(0)
[   11.370000] get_wdev_by_idx: invalid idx(0)
[   11.380000] get_wdev_by_idx: invalid idx(0)
[   11.390000] pci probe count=1
[   11.400000] ip_tables: (C) 2000-2006 Netfilter Core Team
[   11.420000] nf_conntrack version 0.5.0 (2048 buckets, 8192 max)
[   11.460000] xt_time: kernel timezone is -0000
[   11.480000] PPP generic driver version 2.4.2
[   11.490000] NET: Registered protocol family 24
[   11.500000] kmodloader: done loading kernel modules from /etc/modules.d/*
[   14.730000] get_wdev_by_idx: invalid idx(0)
[   17.290000] get_wdev_by_idx: invalid idx(0)
[   17.300000] get_wdev_by_idx: invalid idx(0)
[   17.490000] jffs2_scan_eraseblock(): End of filesystem marker found at 0x0
[   17.500000] jffs2_build_filesystem(): unlocking the mtd device... [   17.520000] done.
[   17.520000] jffs2_build_filesystem(): erasing all blocks after the end marker... [   17.660000] br-lan: port 1(eth0.1) entered blocking state
[   17.670000] br-lan: port 1(eth0.1) entered disabled state
[   17.680000] device eth0.1 entered promiscuous mode
[   17.690000] device eth0 entered promiscuous mode
[   17.700000] br-lan: port 1(eth0.1) entered blocking state
[   17.710000] br-lan: port 1(eth0.1) entered forwarding state
[   18.160000] get_wdev_by_idx: invalid idx(0)
[   18.170000] get_wdev_by_idx: invalid idx(0)
[   21.310000] get_wdev_by_idx: invalid idx(0)
[   21.620000] get_wdev_by_idx: invalid idx(0)
[   67.360000] done.
[   67.360000] jffs2: notice: (855) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
```

981213 commented:
There is a web recovery module in the stock u-boot but it only accept the factory binary.
```
============================================ 
Ralink UBoot Version: 5.0.0.0
-------------------------------------------- 
ASIC MT7621A DualCore (MAC to MT7530 Mode)
DRAM_CONF_FROM: Auto-Detection 
DRAM_TYPE: DDR3 
DRAM bus: 16 bit
Xtal Mode=3 OCP Ratio=1/3
Flash component: SPI Flash
Date:  Time:
============================================ 
icache: sets:256, ways:4, linesz:32 ,total:32768
dcache: sets:256, ways:4, linesz:32 ,total:32768 

 ##### The CPU freq = 880 MHZ #### 
 estimate memory size =128 Mbytes
#Reset_MT7530
set LAN/WAN LLLLW

Press press hardware button for more than 4 seconds to run web failsafe mode

 5

hardware button was pressed for 5 seconds
HTTP server is starting for firmware update...



 NetTxPacket = 0x87FE3480 

 NetRxPackets[0] = 0x87FE3A80

 NetRxPackets[1] = 0x87FE4080

 NetRxPackets[2] = 0x87FE4680

 NetRxPackets[3] = 0x87FE4C80

 NetRxPackets[4] = 0x87FE5280

 NetRxPackets[5] = 0x87FE5880

 NetRxPackets[6] = 0x87FE5E80

 NetRxPackets[7] = 0x87FE6480

 NetRxPackets[8] = 0x87FE6A80

 NetRxPackets[9] = 0x87FE7080

 NetRxPackets[10] = 0x87FE7680

 NetRxPackets[11] = 0x87FE7C80

 NetRxPackets[12] = 0x87FE8280

 NetRxPackets[13] = 0x87FE8880

 NetRxPackets[14] = 0x87FE8E80

 NetRxPackets[15] = 0x87FE9480

 NetRxPackets[16] = 0x87FE9A80

 NetRxPackets[17] = 0x87FEA080

 NetRxPackets[18] = 0x87FEA680

 NetRxPackets[19] = 0x87FEAC80

 NetRxPackets[20] = 0x87FEB280

 NetRxPackets[21] = 0x87FEB880

 NetRxPackets[22] = 0x87FEBE80

 NetRxPackets[23] = 0x87FEC480

 KSEG1ADDR(NetTxPacket) = 0xA7FE3480 

 NetLoopHttpd,call eth_halt ! 
Trying Eth0 (10/100-M)

 Waitting for RX_DMA_BUSY status Start... done


 ETH_STATE_ACTIVE!! 
HTTP server is starting at IP: 192.168.2.1
HTTP server is ready!

Request for: /
Request for: /style.css
Request for: /favicon.ico
## Error: file not found!
Data will be downloaded at 0x84000000 in RAM
Upgrade type: sopimg
Upload file size: 5243048 bytes
Loading: #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################
         #######################################

HTTP upload is done! Upgrading...


****************************
*    SOPIMG UPGRADING    *
* DO NOT POWER OFF DEVICE! *
****************************

Not found the header
## Error: HTTP ugrade failed!


 NetLoopHttpd,call eth_halt ! 
Trying Eth0 (10/100-M)

 ETH_STATE_ACTIVE!! 
HTTP server is starting at IP: 192.168.2.1
HTTP server is ready!

```

mkresin commented:
merged to my [staging tree](https://git.lede-project.org/?p=lede/mkresin/staging.git;a=summary), thanks!
